### PR TITLE
wip: configstore/podclusters has syntax sugar for interacting with PCs

### PR DESCRIPTION
**Pull request by rudle**
PCs are a primary, but not necessarily exclusive use case for config.

### CODE REVIEW

File: pkg/store/consul/configstore/podclusters/podclusters.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	"fmt"
+	"square/up/util"
+
+	pfields "github.com/square/p2/pkg/pc/fields"
+	"github.com/square/p2/pkg/store/consul/configstore"
+	"github.com/square/p2/pkg/types"
+	klabels "k8s.io/kubernetes/pkg/labels"
+)
+
+func sup() {
```
Comment (rudle) [inline pkg/store/consul/configstore/podclusters/podclusters.go]: sup

File: pkg/store/consul/configstore/podclusters/podclusters.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	LabelConfig(configstore.ID, map[string]string) error
+	FetchConfig(configstore.ID) (configstore.Fields, *configstore.Version, error)
+}
+
+type PodClusterConfigStore struct {
+	configStore ConfigStore
+}
+
+// TODO this needs a UUID generator
+func (pccs *PodClusterConfigStore) CreateOrUpdateConfigForPodCluster(ctx context.Context, podID types.PodID, az pfields.AvailabilityZone, cn pfields.ClusterName, fields configstore.Fields) (*configstore.Fields, error) {
```
Comment (mpuncel) [inline pkg/store/consul/configstore/podclusters/podclusters.go]: probably should take a version here too for CAS purposes.

**Comment by mpuncel** [comment]
I ended up using this commit within #900 so probably we should close this one and look at that one

**Comment by rudle** [comment]
Closing as stale

**Pull request closed without merging**